File: modules/mrlocallocal/src/main/java/com/soundcloud/android/mrlocallocal/RealLogger.java
package com.soundcloud.android.mrlocallocal;

import android.util.Log;

class RealLogger implements Logger {
	private static final String TAG = "MrLocalLocal";

	@Override
	public void info(String message) {
		Log.w(TAG, message);
	}

	@Override
	public void error(String message) {
		Log.e(TAG, message);
	}
}

File: modules/mrlocallocal/src/main/java/com/soundcloud/android/mrlocallocal/RealLogger.kt
package com.soundcloud.android.mrlocallocal

import android.util.Log
import java.lang.StringBuilder

internal class RealLogger : Logger {
    companion object {
        private val TAG = "MrLocalLocal"
    }
    private val resultOutput = StringBuilder()

    override fun info(message: String) {
        Log.i(TAG, message)
        resultOutput.appendln(message)
    }

    override fun error(message: String) {
        Log.e(TAG, message)
        resultOutput.appendln(message)
    }

    override fun getResultOutput(): String {
        return resultOutput.toString()
    }
}
